src/app/frontend/common/resource/editresource_controller.js

```diff
@@ -67,11 +75,35 @@ export class EditResourceController {
   }
 
   /**
+   * @export
+   */
+  copy() {
+    /* if the clipboard functionality is not supported */
+    if (!this.clipboard_.supported) {
+      this.showMessage('Your browser does not support clipboard');
+    } else {
+      this.clipboard_.copyText(JSON.stringify(this.data, null, 2));
+      this.showMessage('copied to clipboard');
+    }
+  }
+
+  /**
    * Cancels and closes the dialog.
    *
    * @export
    */
   cancel() {
     this.mdDialog_.cancel();
   }
+
+  /**
+   * show an error message
+   */
+  showMessage(message) {
```

Comment:
Our pattern for private functions is to use `_` suffix. Also please annotate this function with `@private` and `@param`.